user-management/build.gradle

```diff
@@ -0,0 +1,298 @@
+buildscript {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        maven { url "https://repo.spring.io/plugins-release" }
+    }
+    dependencies {
+        classpath "io.spring.gradle:propdeps-plugin:0.0.10.RELEASE"
+    }
+}
+
+plugins {
+    id "checkstyle"
+    id "java"
+    id "maven-publish"
+    id "idea"
+    id "jacoco"
+    id "org.springframework.boot"
+    id "com.google.cloud.tools.jib"
+    id "com.github.node-gradle.node"
+    id "com.diffplug.spotless"
+    // this allows us to find outdated dependencies via ./gradlew dependencyUpdates
+    id "com.github.ben-manes.versions"
+    id "com.github.andygoossens.gradle-modernizer-plugin"
+    id "org.openapi.generator" version "5.1.1" // version is included since the plugin is not used by Artemis
+    id "io.spring.nohttp" version "0.0.8" // version is included since the plugin is not used by Artemis
+    //jhipster-needle-gradle-plugins - JHipster will add additional gradle plugins here
+}
+
+group = "de.tum.in.www1.artemis"
+
+sourceCompatibility=16
+targetCompatibility=16
+
+apply from: "gradle/docker.gradle"
+apply from: "../gradle/liquibase.gradle"
+//jhipster-needle-gradle-apply-from - JHipster will add additional gradle scripts to be applied here
+
+if (project.hasProperty("prod")) {
+    apply from: "gradle/profile_prod.gradle"
+} else {
+    apply from: "gradle/profile_dev.gradle"
+}
+
+if (project.hasProperty("war")) {
+    apply from: "gradle/war.gradle"
+}
+
+if (project.hasProperty("zipkin")) {
+    apply from: "gradle/zipkin.gradle"
+}
+
+
+apply from: "../gradle/tasks.gradle"
+
+private excludedClassFilesForReport(classDirectories) {
+    classDirectories.setFrom(files(classDirectories.files.collect {
+        fileTree(dir: it,
+            includes: ['**/de/tum/in/www1/artemis/usermanagement/**'],
+        )
+    }))
+}
+
+jacocoTestReport.configure {
+    // we want to ignore files different from the current project
+    afterEvaluate { excludedClassFilesForReport(classDirectories) }
+}
+jacocoTestCoverageVerification.configure {
+    // we want to ignore files different from the current project
+    afterEvaluate { excludedClassFilesForReport(classDirectories) }
+}
+
+sourceSets {
+    external {
+        java {
+            srcDir file("../src/main/java")
+                // de.tum.in.www1.artemis.security
+            include 'de/tum/in/www1/artemis/security/DomainUserDetailsService.*'
```

Comment:
What would you consider the advantage here by specifying the files so strictly?
Would it be too bad if we include e.g. all of config?

=> Readability vs small build size probably?

Reply:
Small build size is one of the reasons. Another reason is that I want to make the microservices as little as possible dependent on Artemis so that in the future if decided it is easier to remove those dependencies and make the microservice more independent.
In most cases, if I include the whole package I have to include additional classes that are not related to the current microservice, i.e. if I include the CacheConfiguration, which I wanted to I have to include a dependency to QuizScheduleService and to all of the services this service uses. This from my point of view should be avoided.